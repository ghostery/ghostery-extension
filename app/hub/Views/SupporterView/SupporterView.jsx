/**
 * Supporter View Component
 *
 * Ghostery Browser Extension
 * https://www.ghostery.com/
 *
 * Copyright 2018 Ghostery, Inc. All rights reserved.
 *
 * This Source Code Form is subject to the terms of the Mozilla Public
 * License, v. 2.0. If a copy of the MPL was not distributed with this
 * file, You can obtain one at http://mozilla.org/MPL/2.0
 *
 * ToDo: Update this file.
 */

import React, { Component } from 'react';

/**
 * @class Implement the Supporter View for the Ghostery Hub
 * @extends Component
 * @memberof HubComponents
 */
class SupporterView extends Component {
	constructor(props) {
		super(props);

		const title = t('hub_supporter_page_title');
		window.document.title = title;
	}

	/**
	 * React's required render function. Returns JSX
	 * @return {JSX} JSX for rendering the Supporter View of the Hub app
	 */
	render() {
		return (
			<div className="SupporterView full-height">

				<div className="SupporterView__header row align-center SupporterView--paddingTopSmall">
					<div className="columns">
						<div className="SupporterView__headerImage"/>

						<div className="SupporterView__title SupporterView--centeredText">
							{t('hub_supporter_header_title')}
						</div>

						<div className="SupporterView__description SupporterView--centeredText">
							{t('hub_supporter_header_description')}
						</div>

						<div className="row align-center-middle">

								<div className="button">
									{t('hub_supporter_button_text')}
								</div>

								<span className="SupporterView__priceTag"><sup>$</sup>2</span>
								<span className="SupporterView__priceTagFrequency">per month</span>

						</div>
					</div>
				</div>

				<div className="SupporterView--paddingTopSmall row align-center">
					<div className="SupporterView__perkFeature columns">

						<div className="SupporterView__perkIconThemes" />
						<div className="SupporterView__perkTitle">
							{t('hub_supporter_perk_themes_title')}
						</div>
						<div className="SupporterView__perkDescription">
							{t('hub_supporter_perk_themes_description')}
						</div>

					</div>

					<div className="SupporterView__perkFeature columns">

						<div className="SupporterView__perkIconPriority" />
						<div className="SupporterView__perkTitle">
							{t('hub_supporter_perk_priority_title')}
						</div>
						<div className="SupporterView__perkDescription">
							{t('hub_supporter_perk_priority_description')}
						</div>
					</div>

					<div className="SupporterView__perkFeature columns">
						<div className="SupporterView__perkIconMore" />
						<div className="SupporterView__perkTitle">
							{t('hub_supporter_perk_more_title')}
						</div>
						<div className="SupporterView__perkDescription">
							{t('hub_supporter_perk_more_description')}
						</div>
					</div>
				</div>

				<div className="SupporterView__manifesto row expanded SupporterView--marginTop align-center-middle">
					<div className="SupporterView__manifestoText">
						{t('hub_supporter_manifesto')}
					</div>
				</div>

				<div className="SupporterView__themes row SupporterView--marginTop align-middle">
					<div className="SupporterView__themesText columns">
						<div className="SupporterView__title">
							{t('hub_supporter_themes_title')}
						</div>

						<div className="SupporterView__description">
							{t('hub_supporter_themes_description')}
						</div>

						<div className="button">
							{t('hub_supporter_button_text')}
						</div>
					</div>

					<div className="columns">
						<img src="/app/images/hub/supporter/screenshot.png" className="SupporterView__themesScreenshots" />
					</div>
				</div>


				<div className="SupporterView__priority SupporterView--marginTop row expanded">

					<div className="row align-center-middle">
						<div className="SupporterView__priorityIcon columns" />

						<div className="SupporterView__priorityText columns">

							<div className="SupporterView__title row">
								{t('hub_supporter_priority_title')}
							</div>

							<div className="SupporterView__description row">
								{t('hub_supporter_priority_description')}
							</div>

							<div className="button row">
								{t('hub_supporter_button_text')}
							</div>
						</div>
					</div>

				</div>


				<div className="SupporterView__manyMore row align-center">

					<div className="SupporterView__manyMoreText columns">

						<div className="SupporterView__title SupporterView--centeredText">
							{t('hub_supporter_more_title')}
						</div>

						<div className="SupporterView__description SupporterView--centeredText">
							{t('hub_supporter_more_description')}
						</div>

						<div className="SupporterView__manyMoreIcon" />

					</div>
				</div>

				<div className="SupporterView__footer row align-center-middle SupporterView--paddingTopSmall">
					<div className="SupporterView__footerLine columns" />

					<div className="SupporterView__buttonContainer columns">
						<div className="button">
							{t('hub_supporter_button_text')}
						</div>
					</div>

					<div className="SupporterView__footerLine columns" />

				</div>

			</div>
		);
	}
}

export default SupporterView;
