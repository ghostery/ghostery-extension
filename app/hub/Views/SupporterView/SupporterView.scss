/**
 * Supporter View Sass
 *
 * Ghostery Browser Extension
 * https://www.ghostery.com/
 *
 * Copyright 2018 Ghostery, Inc. All rights reserved.
 *
 * This Source Code Form is subject to the terms of the Mozilla Public
 * License, v. 2.0. If a copy of the MPL was not distributed with this
 * file, You can obtain one at http://mozilla.org/MPL/2.0
 */

// Supporter View
.SupporterView {}
.SupporterView--centeredText {
  text-align: center;
}
.SupporterView--paddingTop {
	padding-top: 80px;
}
.SupporterView--paddingTopSmall {
  padding-top: 40px;
}
.SupporterView--marginTop {
	margin-top: 80px;
}
.SupporterView {
  margin-bottom: 80px;
}
.SupporterView__headerImage {
  max-width: 100%;
  height: 195px;
  margin-right: 30px;
  background: buildIconGoldenGhost(#000000) no-repeat center center;
}
.SupporterView__title {
  margin-top: 40px;
  line-height: 38px;
	font-size: 24px;
	font-weight: 700;
}
.SupporterView__description {
  line-height: 30px;
	font-size: 14px;
}
.SupporterView__priceTag {
  font-size: 64px;
  font-weight: 500;
  margin: 0 10px 0 50px;
}
sup {
  position: relative;
  bottom:0.9em;
  font-size: 30px;
}
.SupporterView__priceTagFrequency {
  color: #9b9b9b;
  font-size: 14px;
}
.SupporterView__perkIconThemes {
  width: 100%;
  height: 100px;
  background: buildIconThemesPerk(#9ECC42) no-repeat center center;
}
.SupporterView__perkIconPriority {
  width: 100%;
  height: 100px;
  background: buildIconPriorityPerk(#9ECC42) no-repeat center center;
}
.SupporterView__perkIconMore {
  width: 100%;
  height: 100px;
  background: buildIconMorePerk(#9ECC42) no-repeat center center;
}
.SupporterView__perkFeature {
	margin: 0 50px;
	font-size: 14px;
	max-width: 250px;
  text-align: center;
}
.SupporterView__perkTitle {
	line-height: 38px;
	font-weight: 700;
}
.SupporterView__perkDescription {
  line-height: 30px;
}
.SupporterView__manifesto {
  background: linear-gradient(246deg, #ecd868, #d4af37);
  width: 100%;
  height: auto;
}
.SupporterView__manifestoText {
	color: #ffffff;
	font-weight: 700;
	font-size: 24px;
	line-height: 40px;
	padding: 50px;
	text-align: center;
	max-width: 700px;
}
.SupporterView__themesText {
  max-width: 300px;
  margin:50px;
}
.SupporterView__themesScreenshots {
  max-width: 85%;
}
.button {
  margin-top: 10px;
  background: linear-gradient(256deg, #9ecc42, #429321);
  border: none;
  width: 170px;
  height: 34px;
  text-transform: none;
}
.button:hover {
  background-color: #429321;
}
.SupporterView__priority {
  background-color: #f7f7f7;
  padding: 50px;
}
.SupporterView__priorityIcon {
  background: buildIconPriority(#000000) no-repeat center center;
  width: 273px;
  height: 100%;
  margin: 0 60px;
}
.SupporterView__priorityText {
  max-width: 300px;
  margin: 0 60px;
}
.SupporterView__manyMoreIcon {
  background: buildIconManyMore(#000000) no-repeat center center;
  width: 100%;
  height: 200px;
}
.SupporterView__manyMoreText {
  max-width: 430px;
}
.SupporterView__footer {
  margin-bottom: 80px;
}
.SupporterView__buttonContainer {
  max-width: 200px;
  margin: 0 50px;
}
.SupporterView__footerLine {
  max-width: 20vw;
  height: 1px;
  border: solid 0.5px #9b9b9b;
}
