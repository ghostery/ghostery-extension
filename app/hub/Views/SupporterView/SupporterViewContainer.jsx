/**
 * Supporter View Container
 *
 * Ghostery Browser Extension
 * https://www.ghostery.com/
 *
 * Copyright 2018 Ghostery, Inc. All rights reserved.
 *
 * This Source Code Form is subject to the terms of the Mozilla Public
 * License, v. 2.0. If a copy of the MPL was not distributed with this
 * file, You can obtain one at http://mozilla.org/MPL/2.0
 */

import React, { Component } from 'react';
import PropTypes from 'prop-types';

import SupporterView from './SupporterView';

/**
 * @class Implement the Supporter View for the Ghostery Hub
 * @extends Component
 * @memberof HubComponents
 */
class SupporterViewContainer extends Component {
	constructor(props) {
		super(props);

		const title = t('hub_supporter_page_title');
		window.document.title = title;

		props.actions.getUser();
	}

	/**
	 * React's required render function. Returns JSX
	 * @return {JSX} JSX for rendering the Supporter View of the Hub app
	 */
	render() {
		const isSupporter = this.props.user && this.props.user.subscriptionsSupporter || false;

		return <SupporterView isSupporter={isSupporter} />;
	}
}

// PropTypes ensure we pass required props of the correct type
SupporterViewContainer.propTypes = {
	user: PropTypes.shape({
		subscriptionsSupporter: PropTypes.bool,
	}),
	actions: PropTypes.shape({
		getUser: PropTypes.func.isRequired,
	}).isRequired,
};

// Default props used in the Supporter View
SupporterViewContainer.defaultProps = {
	user: {
		subscriptionsSupporter: false,
	},
};

export default SupporterViewContainer;
