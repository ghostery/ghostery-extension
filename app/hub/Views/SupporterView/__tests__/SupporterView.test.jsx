/**
 * Supporter View Test Component
 *
 * Ghostery Browser Extension
 * https://www.ghostery.com/
 *
 * Copyright 2018 Ghostery, Inc. All rights reserved.
 *
 * This Source Code Form is subject to the terms of the Mozilla Public
 * License, v. 2.0. If a copy of the MPL was not distributed with this
 * file, You can obtain one at http://mozilla.org/MPL/2.0
 */

import React from 'react';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';
import SupporterView from '../SupporterView';

describe('app/hub/Views/SupporterView component', () => {
	describe('Snapshot tests with react-test-renderer', () => {
		test('rewards view is rendered correctly', () => {
			const component = renderer.create(
				<SupporterView />
			).toJSON();
			expect(component).toMatchSnapshot();
		});
	});

	describe('Shallow snapshot tests rendered with Enzyme', () => {
		test('the happy path of the component', () => {
			const component = shallow(<SupporterView />);
			expect(component.find('.SupporterView').length).toBe(1);
			expect(component.find('.button').length).toBe(4);
			expect(component.find('.SupporterView__perkFeature').length).toBe(3);
			expect(component.find('.SupporterView__title').length).toBe(4);
			expect(component.find('.SupporterView__description').length).toBe(4);
		});
	});
});
