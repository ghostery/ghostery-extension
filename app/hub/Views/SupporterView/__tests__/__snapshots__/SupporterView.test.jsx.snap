// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`app/hub/Views/SupporterView component Snapshot tests with react-test-renderer rewards view is rendered correctly 1`] = `
<div
  className="SupporterView full-height"
>
  <div
    className="SupporterView__header row align-center SupporterView--paddingTopSmall"
  >
    <div
      className="columns"
    >
      <div
        className="SupporterView__headerImage"
      />
      <div
        className="SupporterView__title SupporterView--centeredText"
      >
        hub_supporter_header_title
      </div>
      <div
        className="SupporterView__description SupporterView--centeredText"
      >
        hub_supporter_header_description
      </div>
      <div
        className="row align-center-middle"
      >
        <div
          className="button"
        >
          hub_supporter_button_text
        </div>
        <span
          className="SupporterView__priceTag"
        >
          <sup>
            $
          </sup>
          2
        </span>
        <span
          className="SupporterView__priceTagFrequency"
        >
          per month
        </span>
      </div>
    </div>
  </div>
  <div
    className="SupporterView--paddingTopSmall row align-center"
  >
    <div
      className="SupporterView__perkFeature columns"
    >
      <div
        className="SupporterView__perkIconThemes"
      />
      <div
        className="SupporterView__perkTitle"
      >
        hub_supporter_perk_themes_title
      </div>
      <div
        className="SupporterView__perkDescription"
      >
        hub_supporter_perk_themes_description
      </div>
    </div>
    <div
      className="SupporterView__perkFeature columns"
    >
      <div
        className="SupporterView__perkIconPriority"
      />
      <div
        className="SupporterView__perkTitle"
      >
        hub_supporter_perk_priority_title
      </div>
      <div
        className="SupporterView__perkDescription"
      >
        hub_supporter_perk_priority_description
      </div>
    </div>
    <div
      className="SupporterView__perkFeature columns"
    >
      <div
        className="SupporterView__perkIconMore"
      />
      <div
        className="SupporterView__perkTitle"
      >
        hub_supporter_perk_more_title
      </div>
      <div
        className="SupporterView__perkDescription"
      >
        hub_supporter_perk_more_description
      </div>
    </div>
  </div>
  <div
    className="SupporterView__manifesto row expanded SupporterView--marginTop align-center-middle"
  >
    <div
      className="SupporterView__manifestoText"
    >
      hub_supporter_manifesto
    </div>
  </div>
  <div
    className="SupporterView__themes row SupporterView--marginTop align-middle"
  >
    <div
      className="SupporterView__themesText columns"
    >
      <div
        className="SupporterView__title"
      >
        hub_supporter_themes_title
      </div>
      <div
        className="SupporterView__description"
      >
        hub_supporter_themes_description
      </div>
      <div
        className="button"
      >
        hub_supporter_button_text
      </div>
    </div>
    <div
      className="columns"
    >
      <img
        className="SupporterView__themesScreenshots"
        src="/app/images/hub/supporter/screenshot.png"
      />
    </div>
  </div>
  <div
    className="SupporterView__priority SupporterView--marginTop row expanded"
  >
    <div
      className="row align-center-middle"
    >
      <div
        className="SupporterView__priorityIcon columns"
      />
      <div
        className="SupporterView__priorityText columns"
      >
        <div
          className="SupporterView__title row"
        >
          hub_supporter_priority_title
        </div>
        <div
          className="SupporterView__description row"
        >
          hub_supporter_priority_description
        </div>
        <div
          className="button row"
        >
          hub_supporter_button_text
        </div>
      </div>
    </div>
  </div>
  <div
    className="SupporterView__manyMore row align-center"
  >
    <div
      className="SupporterView__manyMoreText columns"
    >
      <div
        className="SupporterView__title SupporterView--centeredText"
      >
        hub_supporter_more_title
      </div>
      <div
        className="SupporterView__description SupporterView--centeredText"
      >
        hub_supporter_more_description
      </div>
      <div
        className="SupporterView__manyMoreIcon"
      />
    </div>
  </div>
  <div
    className="SupporterView__footer row align-center-middle SupporterView--paddingTopSmall"
  >
    <div
      className="SupporterView__footerLine columns"
    />
    <div
      className="SupporterView__buttonContainer columns"
    >
      <div
        className="button"
      >
        hub_supporter_button_text
      </div>
    </div>
    <div
      className="SupporterView__footerLine columns"
    />
  </div>
</div>
`;
